addons/udes_stock/models/res_users.py

```diff
@@ -37,3 +67,59 @@ def get_user_warehouse(self, aux_domain=None):
                 )
             )
         return warehouse
+
+    def assign_picking_to_users(self, picking):
+        """
+        Assign a picking to user(s) in self
+        """
+        picking.ensure_one()
+        picking_type = picking.picking_type_id
+
+        # Find the users to assign
+        users_to_assign = self.filtered(lambda u: u.u_picking_id != picking)
+        if users_to_assign:
+            if picking.state in NON_ASSIGNABLE_STATES:
+                _logger.warning(
+                    "Cannot assign users %s to picking %s as it is in state %s"
```

Comment:
We should probably wrap the message with `_(...)` here and in the else branch.
Using the `%` operator here prevents lazy evaluation of the final message; 
```python
_logger.warning(_('..%s ... %s'), 'spam', 'ham')
```
would be better.

Reply:
We're still using the `%` operator?
```python
                _logger.warning(
                    _("Cannot assign users %s to picking %s as picking type %s only allows one user"),
                   (self.mapped("name"), picking.name, picking_type.name)
                )
```
is better; if the log message is not emitted the complete string will never be constructed.

Reply:
I thought was translatable thing as the % operator it is used widely in odoo core UserErrors etc

Reply:
UserErrors are a different thing.

If I do `s = "spam %s' % 'eggs'" then the formatting is executed immediately - s is always 'spam eggs'.

The logging package does not behave in quite the same way as normal code.  With a logging statement like `_logger.info('spam %s', 'eggs')`, the logging package computes whether or not the message will be emitted, and only executes the formatting if the message will be emitted.  However if we do `_logger.info('spam %s' % 'eggs')` then the formatting is evaluated _before_ the logging package determines whether the message will ever be logged.  So not using `%` in this way is a (very small) efficiency gain.